src/client/oicp/CpoOICPClient.ts

```diff
@@ -0,0 +1,1420 @@
+import ChargingStation, { Connector, RemoteAuthorization } from '../../types/ChargingStation';
```

Comment:
Yes, there is potential to share the algorithms.
I will check it over time. But since this is not critical for the merge, I will to it over time.

I want to prioritize the merge and tests